content/guides/self-hosting-umami-neon.md

```diff
@@ -0,0 +1,202 @@
+
+- [flyctl](https://fly.io/docs/getting-started/installing-flyctl/) – command-line utility that lets you work with the Fly.io platform. You will also need [a fly.io account](https://fly.io/docs/hands-on/sign-up/).
+- [A Neon account](https://console.neon.tech/signup) – The self-hosted Umami analytics instance will connect to a Neon serverless Postgres database 🚀
+
+## Steps
+
+- [What is Umami?](#what-is-umami)
+- [Provisioning a Postgres database using Neon](#provisioning-a-postgres-database-using-neon)
+- [Setup Umami instance for Fly.io](#setup-umami-instance-for-flyio)
+- [Set Neon Postgres as Serverless Database for Self-Hosted Umami Analytics](#set-neon-postgres-as-serverless-database-for-self-hosted-umami-analytics)
+- [Deploy To Fly.io](#deploy-to-flyio)
+
+## What is Umami?
+
+![Umami Analytics Preview](/guides/images/self-hosting-umami-neon/umami.jpeg)
+
+Umami is a simple, fast, privacy-focused, open-source analytics solution. Umami is a better alternative to Google Analytics because it gives you total control of your data and does not violate the privacy of your users. <sup>[[1](https://umami.is/docs)]</sup>
+
+## Provisioning a Postgres Database using Neon
+
+Using Serverless Postgres database powered by Neon helps you scale down to zero. With Neon, you only have to pay for what you use.
+
+To get started, go to the [Neon console](https://console.neon.tech/app/projects) and enter the name of your choice as the project name.
+
+You will then be presented with a dialog that provides a connecting string of your database. Click on **Pooled connection** on the top right of the dialog and the connecting string automatically updates in the box below it.
+
+![](/guides/images/self-hosting-umami-neon/1689d44f-4c5d-4b2a-8d13-32407f9c8781.png)
+
+All Neon connection strings have the following format:
+
+```bash
+postgres://<user>:<password>@<endpoint_hostname>.neon.tech:<port>/<dbname>
+```
+
+- `user` is the database user.
+- `password` is the database user’s password.
+- `endpoint_hostname` is the host with neon.tech as the [TLD](https://www.cloudflare.com/en-gb/learning/dns/top-level-domain/).
+- `port` is the Neon port number. The default port number is 5432.
+- `dbname` is the name of the database. “neondb” is the default database created with each Neon project.
+- `?sslmode=require` an optional query parameter that enforces the [SSL](https://www.cloudflare.com/en-gb/learning/ssl/what-is-ssl/) mode while connecting to the Postgres instance for better security.
+
+Save this connecting string somewhere safe to be used as the `DATABASE_URL` further as we proceed.
+
+## Setup Umami analytics instance for Fly.io
+
+To self host your Umami analytics instance, you would use [Umami's pre-built Docker container for Postgres](https://github.com/umami-software/umami/pkgs/container/umami/157800125?tag=postgresql-latest). This will allow you to self host an Umami analytics instance on Fly.io with a single `fly.toml` file.
```

Comment:
```suggestion
To self host your Umami analytics instance, you'll use [Umami's pre-built Docker container for Postgres](https://github.com/umami-software/umami/pkgs/container/umami/157800125?tag=postgresql-latest). This will allow you to self host an Umami analytics instance on Fly.io with a single `fly.toml` file.
```